src/jira.py

```diff
@@ -133,14 +141,20 @@ def get_utc_offset(timestamp):
         return datetime.timezone(datetime.timedelta(seconds=int(offset_direction + str(offset_seconds))))
 
     def get_github_equivalent(self):
-        """Find the equivalent Github issue key and repo name if listed in the issue text. Issues synced by unito-bot
-        will have this information."""
-
+        """Find the equivalent Github issue key, repository name, milestone name and number if listed in the
+        description field. Issues synchronized by Unito will have this information, but not all issue descriptions
+        have the corresponding GitHub issue listed in them."""
         if self.description:
-            match_obj = re.search(r'Repository Name: ([\w_-]*)[\s\S]*Issue Number: ([\w-]*)', self.description)
-            if match_obj:
-                return match_obj.group(1), match_obj.group(2)
-            print(self.jira_key, 'No match was found in the description.')
+            match_obj1 = re.search(r'(?<=Repository Name: )(.*?)(?={color})', self.description)
+            match_obj2 = re.search(r'(?<=Issue Number: )(.*?)(?={color})', self.description)
+            match_obj3 = re.search(r'(?<=Milestone: )(.*?)(?={color})', self.description)
+            match_obj4 = re.search(r'(?<=github.com/)(.*?)(?=/)', self.description)
+            if not any([match_obj1, match_obj2, match_obj3, match_obj4]):
+                print('No match was found in the description.')
+            self.github_repo = match_obj1.group(0) if match_obj1 else None
+            self.github_key = match_obj2.group(0) if match_obj2 else None
+            self.github_milestone = match_obj3.group(0) if match_obj3 else None
+            self.github_org = match_obj4.group(0) if match_obj4 else None
```

Comment:
could this be condensed into just one field for self.github_issue_object? each issue object should know its repo, key, milestone, and org

Reply:
You mean that `self.github_issue_object` contains a dictionary? We can do that. It would, however, deviate from the pattern of used in the other, similar-looking classes such as `issue`, `zenhub`, etc. Plus I don't think the code gains in efficiency by doing this, and/or readability. Thoughts?
Having said that, I agree that the multiple lines of matches is ugly, but I haven't figured out how to use multiple look-ahead patterns in one regex...